data/transactions/logic/backwardCompat_test.go

```diff
@@ -280,98 +277,71 @@ func TestBackwardCompatTEALv1(t *testing.T) {
 		Data:        data[:],
 	})
 
-	txn := makeSampleTxn()
+	ep, tx, _ := makeSampleEnvWithVersion(1)
 	// RekeyTo disallowed on TEAL v0/v1
-	txn.Txn.RekeyTo = basics.Address{}
-	txgroup := makeSampleTxnGroup(txn)
-	txn.Lsig.Logic = program
-	txn.Lsig.Args = [][]byte{data[:], sig[:], pk[:], txn.Txn.Sender[:], txn.Txn.Note}
-	txn.Txn.RekeyTo = basics.Address{} // RekeyTo not allowed in TEAL v1
+	tx.RekeyTo = basics.Address{}
 
-	sb := strings.Builder{}
-	ep := defaultEvalParamsWithVersion(&sb, &txn, 1)
-	ep.TxnGroup = txgroup
+	ep.TxnGroup[0].Lsig.Logic = program
+	ep.TxnGroup[0].Lsig.Args = [][]byte{data[:], sig[:], pk[:], tx.Sender[:], tx.Note}
 
 	// ensure v1 program runs well on latest TEAL evaluator
 	require.Equal(t, uint8(1), program[0])
 
 	// Cost should stay exactly 2140
 	ep.Proto.LogicSigMaxCost = 2139
-	err = Check(program, ep)
+	err = CheckSignature(0, ep)
 	require.Error(t, err)
 	require.Contains(t, err.Error(), "static cost")
 
 	ep.Proto.LogicSigMaxCost = 2140
-	err = Check(program, ep)
+	err = CheckSignature(0, ep)
 	require.NoError(t, err)
 
-	pass, err := Eval(program, ep)
+	pass, err := EvalSignature(0, ep)
 	if err != nil || !pass {
 		t.Log(hex.EncodeToString(program))
-		t.Log(sb.String())
+		t.Log(ep.Trace.String())
 	}
 	require.NoError(t, err)
 	require.True(t, pass)
 
 	// Costs for v2 should be higher because of hash opcode cost changes
-	ep2 := defaultEvalParamsWithVersion(&sb, &txn, 2)
-	ep2.TxnGroup = txgroup
+	ep2, tx, _ := makeSampleEnvWithVersion(2)
 	ep2.Proto.LogicSigMaxCost = 2307
-	err = Check(opsV2.Program, ep2)
-	require.Error(t, err)
-	require.Contains(t, err.Error(), "static cost")
+	//	ep2.TxnGroup[0].Lsig.Logic = opsV2.Program
+	ep2.TxnGroup[0].Lsig.Args = [][]byte{data[:], sig[:], pk[:], tx.Sender[:], tx.Note}
+	// Eval doesn't fail, but it would be ok (better?) if it did
+	testLogicBytes(t, opsV2.Program, ep2, "static cost", "")
 
 	ep2.Proto.LogicSigMaxCost = 2308
-	err = Check(opsV2.Program, ep2)
-	require.NoError(t, err)
-
-	pass, err = Eval(opsV2.Program, ep2)
-	if err != nil || !pass {
-		t.Log(hex.EncodeToString(ops.Program))
-		t.Log(sb.String())
-	}
-	require.NoError(t, err)
-	require.True(t, pass)
+	testLogicBytes(t, opsV2.Program, ep2)
 
 	// ensure v0 program runs well on latest TEAL evaluator
-	ep = defaultEvalParams(&sb, &txn)
-	ep.TxnGroup = txgroup
+	ep, tx, _ = makeSampleEnv()
 	program[0] = 0
 	sig = c.Sign(Msg{
 		ProgramHash: crypto.HashObj(Program(program)),
 		Data:        data[:],
 	})
-	txn.Lsig.Logic = program
-	txn.Lsig.Args = [][]byte{data[:], sig[:], pk[:], txn.Txn.Sender[:], txn.Txn.Note}
+	ep.TxnGroup[0].Lsig.Logic = program
+	ep.TxnGroup[0].Lsig.Args = [][]byte{data[:], sig[:], pk[:], tx.Sender[:], tx.Note}
 
 	// Cost remains the same, because v0 does not get dynamic treatment
 	ep.Proto.LogicSigMaxCost = 2139
-	err = Check(program, ep)
-	require.Error(t, err)
+	ep.MinTealVersion = new(uint64) // Was higher because sample txn has a rekey
```

Comment:
The evaluator calculates a MinTealVersion is none is supplied upon first access.  I wanted to remove that because I found it surprising, but that would require that we change a lot of callers for LogicSig evaluation that assume they can set up an EvalParams by hand (which leaves MinTealVersion = nil) and go ahead.

Reply:
Or are you asking why the old test did not need to have the mintealversion changed?  Because back on line 289, the old code manipulated the tx RekeyTo field by hand (resetting it).